diff --git a/node_modules/@miniflare/http-server/dist/src/index.js b/node_modules/@miniflare/http-server/dist/src/index.js
index ff84f7a..59af465 100644
--- a/node_modules/@miniflare/http-server/dist/src/index.js
+++ b/node_modules/@miniflare/http-server/dist/src/index.js
@@ -26,7 +26,6 @@ import {
 import { randomHex } from "@miniflare/shared";
 import { coupleWebSocket } from "@miniflare/web-sockets";
 import { Headers } from "undici";
-import { WebSocketServer } from "ws";
 
 // packages/http-server/src/helpers.ts
 import { networkInterfaces } from "os";
@@ -535,6 +534,13 @@ async function createServer(mf, options) {
   } else {
     server = http.createServer(options ?? {}, listener);
   }
+
+  // Fix for Jest :(, Jest 27 doesn't support the "exports" field in
+  // package.json, so incorrectly loads the CommonJS ws module, which includes
+  // WebSocketServer in the default export, not as a named export.
+  const ws = await import("ws");
+  const WebSocketServer = ws.WebSocketServer ?? ws.default.WebSocketServer;
+
   const webSocketServer = new WebSocketServer({ noServer: true });
   const liveReloadServer = new WebSocketServer({ noServer: true });
   server.on("upgrade", async (request, socket, head) => {
